syntax = "proto3";

package protos.event;
option go_package = "./protos/event";

message Value {
  // The value is one of the listed fields. It is valid for all values to be unspecified
  // in which case this AnyValue is considered to be "empty".
  oneof value {
    string string_value = 1;
    bool bool_value = 2;
    int64 int_value = 3;
    double double_value = 4;
    ArrayValue array_value = 5;
    KeyValueList kvlist_value = 6;
    bytes bytes_value = 7;
  }
}

message ArrayValue {
  // Array of values. The array may be empty (contain 0 elements).
  repeated Value values = 1;
}

message KeyValueList {
  // A collection of key/value pairs. The list may be empty (may
  // contain 0 elements).
  // The keys MUST be unique (it is not allowed to have more than one
  // value with the same key).
  repeated KeyValue values = 1;
}

message KeyValue {
  string key = 1;
  Value value = 2;
}

message IngestionEvent {
  string name = 1;
  repeated KeyValue kvs = 2;
//  Timestamp in milliseconds
  uint64 timestamp = 3;
}

message IngestionEventPayload {
  repeated IngestionEvent events = 1;
}

message IngestionEventPayloadRequest {
  IngestionEventPayload data = 1;
}

message IngestionEventPayloadResponse {
  uint64 count = 1;
  string errorMessage = 2;
}

message AWSKinesisEventPayload {
  string data = 1;
}

message AWSKinesisIngestionStreamPayloadRequest {
  string requestId = 1;
  uint64 timestamp = 2;
  repeated AWSKinesisEventPayload records = 3;
}

message AWSKinesisIngestionStreamPayloadResponse {
  string requestId = 1;
  uint64 timestamp = 2;
  string errorMessage = 3;
}
